micrometer/src/main/java/feign/micrometer/MeteredClient.java

```diff
@@ -45,38 +45,82 @@ public MeteredClient(Client client, MeterRegistry meterRegistry, MetricName metr
-              Tag.of("status_group", e.status() / 100 + "xx")))
-          .increment();
+      countErrorResponseCode(template, options, e.status(), e);
       throw e;
     } catch (IOException | RuntimeException e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw e;
     } catch (Exception e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw new IOException(e);
     }
   }
 
+  private void countSuccessResponseCode(RequestTemplate template,
+                                        Options options,
+                                        int responseStatus) {
+    final List<Tag> tags = extraSuccessTags(template, options);
+    tags.add(Tag.of("http_status", String.valueOf(responseStatus)));
+    tags.add(Tag.of("status_group", responseStatus / 100 + "xx"));
+    final Tags allTags = metricName
+        .tag(template.methodMetadata(), template.feignTarget(), tags.toArray(new Tag[] {}));
+    meterRegistry.counter(
+        metricName.name("http_response_code"),
+        allTags)
+        .increment();
+  }
+
+  private void countErrorResponseCode(RequestTemplate template,
+                                      Options options,
+                                      int responseStatus,
+                                      Exception e) {
+    final List<Tag> tags = extraExceptionTags(template, options, e);
+    tags.add(Tag.of("http_status", String.valueOf(responseStatus)));
+    tags.add(Tag.of("status_group", responseStatus / 100 + "xx"));
+    final Tags allTags = metricName
+        .tag(template.methodMetadata(), template.feignTarget(), tags.toArray(new Tag[] {}));
+    meterRegistry.counter(
+        metricName.name("http_response_code"),
+        allTags)
+        .increment();
+  }
+
+  private Timer createSuccessTimer(RequestTemplate template, Options options) {
+    final List<Tag> successTags = extraSuccessTags(template, options);
+    final Tag[] tags = successTags.toArray(new Tag[] {});
+    final Tags allTags = metricName.tag(template.methodMetadata(), template.feignTarget(), tags);
+    return meterRegistry.timer(metricName.name(), allTags);
+  }
+
+  private Timer createExceptionTimer(RequestTemplate template,
+                                     Options options,
+                                     Exception exception) {
+    final List<Tag> exceptionTags = extraExceptionTags(template, options, exception);
+    final Tag[] tags = exceptionTags.toArray(new Tag[] {});
+    final Tags allTags = metricName.tag(template.methodMetadata(), template.feignTarget(), tags);
+    return meterRegistry.timer(metricName.name("exception"), allTags);
+  }
+
+  protected List<Tag> extraSuccessTags(RequestTemplate template, Options options) {
+    final List<Tag> result = new ArrayList<>();
+    result.add(Tag.of("uri", template.path()));
+    return result;
+  }
+
+  protected List<Tag> extraExceptionTags(RequestTemplate template,
```

Comment:
I think we should join together the "success" and "exception"  methods...  I think is ok to have the last argument nullable and only add the tag when needed

Reply:
Done.